atlasdb-feign/src/main/java/com/palantir/atlasdb/http/PollingRefreshable.java

```diff
@@ -46,50 +49,58 @@
     private final Duration refreshInterval;
     private final ScheduledExecutorService poller;
 
-    private final Refreshable<T> refreshable = Refreshable.empty();
+    private final Function<T, U> transform;
+    private final Refreshable<U> refreshable = Refreshable.empty();
 
-    private T lastSeenValue;
+    private T lastSeenValue = null;
 
     private PollingRefreshable(Supplier<T> supplier,
             Duration refreshInterval,
+            Function<T, U> transform,
             ScheduledExecutorService poller) {
         Preconditions.checkArgument(!refreshInterval.isNegative() && !refreshInterval.isZero(),
                 "Refresh interval must be positive, but found %s", refreshInterval);
 
         this.supplier = supplier;
         this.refreshInterval = refreshInterval;
+        this.transform = transform;
         this.poller = poller;
 
         try {
             lastSeenValue = supplier.get();
```

Comment:
Also, I think refreshable#get shouldn't really be called on construction.

Reply:
Hmm, yeah that seems reasonable, though not that keen on changing existing behaviour (especially since the new approach following @OStevan's suggestion doesn't really touch this class)